.devcontainer/README.md

```diff
@@ -2,6 +2,40 @@
 
 For a comprehensive developer environnment supporting AGW c / c++ workflows (extensible with ~effort to other Magma uses) - see the instructions below.
 
+## Prerequisites and Restrictions
+
+DevContainer is a technology developed by Microsoft. Thus, it is only available for [_VS Code_](https://code.visualstudio.com/docs/remote/containers), but **not** for _IntelliJ_. However, as Github is also currently owned by Microsoft, Github provides a DevContainer integration named ["Codespaces"](https://docs.github.com/en/codespaces).
+In order to use the Github codespaces, you must be member of the Magma Github organisation.
+In any case you will need a working Docker installation, as DevContainer is a docker based technology.
+
+Prerequisites in Summary:
+* Local usage:
+  * Docker installed
+  * VS code installed
+* Cloud usage: (=Github codespaces)
+  * membership of Magma organisation
+
+## Developing DevContainer
+
+The DevContainer is a docker image built from the docker file by convention situated at [`.devcontainer/Dockerfile`](Dockerfile).
+The configuration how a DevContainer environment should be started can be found in the [`.devcontainer/devcontainer.json`](devcontainer.json).
+
+Amongst others you have two options to start a DevContainer,
+* either from a prebuilt docker image, e.g. pulled from a container registry with the following configuration node,
+```json
+	"image": "ghcr.io/magma/magma/devcontainer:latest",
+```
+where you will need to have read access to the respective Docker registry, or
+* you can build the docker image anew, when starting the container using the following configuration node instead,
+```json
+	"build": {
+        "dockerfile": "Dockerfile",
+        "context": "..",
+    },
+```
+which requires a connection to the internet to download the respective dependencies. It is important to note that the default context for the `build:` block in the `devcontainer.json` is the `.devcontainer` folder below the repository root. Hence, you must set the context next to the dockerfile to be able to resolve all dependency imports properly.
+If you would want to build the dockerfile manually from the console, you would start the command `docker build -f .devcontainer/Dockerfile .` in the repository root.
+
 ## Setup
```

Comment:
I think some of the doc below might be outdated, do you mind scanning through those as well?

Reply:
Yes, I can have a look there as well. I might not touch the Mac specific stuff though, because I am on a Linux system. :innocent: